.github/workflows/ci.yaml

```
@@ -0,0 +1,155 @@
+
+  test:
+    name: Test Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    needs:
+      - build
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Upgrade snapd
+        run: |
+          sudo snap refresh snapd
+
+      - name: Setup the required system configs OpenSearch
+        run: |
+          sudo snap install opensearch --channel=2/candidate
+          sudo snap connect opensearch:process-control
+
+          sudo sysctl -w vm.swappiness=0
+          sudo sysctl -w vm.max_map_count=262144
+          sudo sysctl -w net.ipv4.tcp_retries2=5
+
+      - name: Connect required interfaces for OpenSearch
+        run: |
+          sudo snap connect opensearch:log-observe
+          sudo snap connect opensearch:mount-observe
+          sudo snap connect opensearch:process-control
+          sudo snap connect opensearch:system-observe
+          sudo snap connect opensearch:sys-fs-cgroup-service
+          sudo snap connect opensearch:shmem-perf-analyzer
+
+      - name: Setup and Start OpenSearch
+        run: |
+          # create the certificates
+          sudo snap run opensearch.setup \
+              --node-name cm0 \
+              --node-roles cluster_manager,data \
+              --tls-priv-key-root-pass root1234 \
+              --tls-priv-key-admin-pass admin1234 \
+              --tls-priv-key-node-pass node1234 \
+              --tls-init-setup yes                 # this creates the root and admin certs as well.
+
+          # start opensearch
+          sudo snap start opensearch.daemon
+
+          # wait a bit for it to fully initialize
+          sleep 15s
+
+          # create the security index
+          sudo snap run opensearch.security-init --tls-priv-key-admin-pass=admin1234
+
+      - name: Download snap file for OpenSearch Dashboards
+        uses: actions/download-artifact@v3
+        with:
+          name: opensearch-dashboards_snap_amd64
+          path: .
+
+      - name: Install snap file for Opensearch Dashboards
+        run: |
+          version="$(cat snap/snapcraft.yaml | yq .version)"
+          
+          sudo snap remove --purge opensearch-dashboards
+          sudo snap install opensearch-dashboards_${version}_amd64.snap --dangerous --jailmode
+
+      - name: Test Opensearch Dashboards
+        run: |
+          sudo snap start opensearch-dashboards
+
+          # Wait until the service esteblishes
+          sleep 15
+
+          # Attempt to access the interface using valid authentication
+          authenticate_ok=$(curl -i  -k -XPOST http://localhost:5601/auth/login \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H 'osd-xsrf:true' \
+            -d '{"username":"kibanaserver","password":"kibanaserver"}' | grep "200 OK") 
+
+          if [ ! "$authenticate_ok" ]; then
+            exit 1
+          fi
```

Please always write a clear error message to STDERR when you are failing. It simplies troubleshooting A LOT!
```suggestion
          if [ -z "${authenticate_ok:-}" ]; then
            echo "ERROR: Authentication failed with correct credentials. Aborting..." >&2
            exit 1
          fi
```